src/coreclr/src/vm/comdelegate.cpp

```diff
@@ -1146,6 +1146,9 @@ PCODE COMDelegate::ConvertToCallback(MethodDesc* pMD)
     }
     CONTRACTL_END;
 
+    // Check the method for proper usage.
+    ThrowIfInvalidUnmanagedCallersOnlyUsage(pMD);
```

Comment:
This method can be used on methods that are not `UnmanagedCallersOnly`. (Notice that we check for it later.)

Reply:
Yep. Agreed. But this version of `ConvertToCallback()` is only ever used for this scenario. I confirmed all cases when I originally exposed `NativeCallableAttribute`. In fact, it only compiles for x86 because of that. I was going to rename it specifically about this scenario so we remember to delete it once x86 is "fixed".

Reply:
Yes, you are right. This is used for `UnmanagedCallersOnly` only.